### PR TITLE
refactor(flags): use kea-forms on feature flags edit page

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Added [kea-forms](https://github.com/keajs/kea-forms) to the system
- Refactored feature flags edit page to use kea-forms
- Rewrote the storybook "[how to build a form](https://61260360a8e859003a7c138b-hcwykbkjow.chromatic.com/?path=/docs/forms-how-to-build-a-form--page)" page with new instructions.
- Fixed various small things in kea-forms along the way, released 0.2.1.
- The aim is to get to reusable and clean form components, but that's for another day. The current hacked together `<Field />` with an antd-imitation template will have to do for now.

## How did you test this code?

- In the browser, I created, edited and listed various feature flags. Going back/forward between them and making edits.

## Next steps

- ~~Use the knowledge learned here to update the "[how to build a form](https://storybook.posthog.net/?path=/docs/forms-how-to-build-a-form--page)" page in storybook~~
- Update some other scenes in the system, try to get rid of all antd forms. There are 14 of them.
- Add the feature flag list and edit pages to storybook

### CODE REVIEW

File: frontend/src/scenes/feature-flags/FeatureFlag.tsx [hunk 1]
```diff
@@ -90,64 +77,39 @@ export function FeatureFlag(): JSX.Element {
...
-                                                </span>
-                                            )
-                                        }}
-                                    </Form.Item>
-                                    <Form.Item name="active" noStyle valuePropName="checked">
-                                        <Switch />
-                                    </Form.Item>
-                                </Form.Item>
-                                {featureFlagId !== 'new' && (
+                                <Field name="active">
```
Comment (pauldambra) [inline frontend/src/scenes/feature-flags/FeatureFlag.tsx:97]: I think I'd approve just for the change of language from `Form.Item` to `Field` 🥇

**Review by pauldambra — CHANGES_REQUESTED** [review]
Two things need fixing IMO

1) you are warned about changing the key when adding one. should only be when changing the key
2) the form values aren't reset when saving (or loading a new flag form)

**Comment by mariusandra** [comment]
1. fixed
2. weird, seems like [this is broken](https://github.com/PostHog/posthog/blob/17290ebf7ceee71bb3f3195520a061cbbec3ccf5/frontend/src/scenes/feature-flags/featureFlagLogic.ts#L357). will check, but first lunch

File: frontend/src/scenes/feature-flags/FeatureFlag.tsx [hunk 2]
```diff
@@ -90,64 +77,39 @@ export function FeatureFlag(): JSX.Element {
...
-                            filters: featureFlag.filters,
-                        })
-                    }
-                    requiredMark={false}
-                    scrollToFirstError
+                    logic={featureFlagLogic}
+                    props={logicProps}
+                    formKey="featureFlag"
+                    className="ant-form-vertical ant-form-hide-required-mark"
+                    // scrollToFirstError
```
Comment (Twixes) [inline frontend/src/scenes/feature-flags/FeatureFlag.tsx:91]: Don't think we need this?

**Review by Twixes — COMMENTED** [review]
What Paul said, overall looks good.

**Comment by Twixes** [comment]
The warning about changing the key during feature flag creation is still a problem for me though. 🤔

**Comment by mariusandra** [comment]
Forgot to push, oops... and also fixed the issue with the new form not reloading.

Basically, `parseInt('new') === NaN !== 'new'`.

**Comment by mariusandra** [comment]
This is how it works now:

**Review by Twixes — APPROVED** [review]
👌

**Pull request merged**